src/ProcessSlave.php

```diff
@@ -201,8 +199,7 @@ function ($item) {
         }
         if ($this->loop) {
             $this->sendCurrentFiles();
-            $this->loop->tick();
-            @$this->loop->stop();
```

Comment:
@andig I was also worried about (tick) plus I had to remove the error suppression (@) to get phpstan on php 7.1 to succeed.

`Result of method React\EventLoop\LoopInterface::stop() (void) is used.`

I think it is a bug in phpstand which this PR fixed https://github.com/phpstan/phpstan/issues/926

Reply:
I'm not worried about phpstan. I'm wondering if the following code is even executed with the tick() removed?

Reply:
ping @TBK could you verify that the code after the tick really gets executed? I'd be happy to merge when confirmed. Thanks!

Reply:
@andig Can confirm that that won't just work like that.

I'm not entirely sure why you have these ticks in here, but the correct workaround would be something like this (using `ProcessManager::shutdown` as an example):

```
if ($this->loop) {
            $this->loop->futureTick(function() {
                $this->loop->stop();

                unlink($this->pidfile);
                exit;
            });
        } else {
            unlink($this->pidfile);
            exit;
        }
```

Calling `futureTick` as-is will never hit the code within the callback because the script exits before it ticks.

Reply:
To clarify: that's also their suggested workaround. All shutdown logic should be _inside_ the callback.

Reply:
So: either the PR needs to be updated or we don't care for the next tick to be executed and stop the loop immediately. To be honest I'm not sure how much value it is to wait for another loop tick as the shutdown- at least is case of hard shutdown- is going to happen anyway?

Reply:
I'm not sure why that tick was in there before either; I'm guessing it was synchronous so it would wait a tick for whatever reason?

Since all the slaves are handled separately I think you're right and the tick may not be necessary. Even for a restart I don't think there's any value in spending that one more tick, as the restart as a whole will be blocked anyway by watching the various PIDs.

Reply:
The only reason we're doing is is apparently `sendCurrentFiles`. I'm unsure why/what for this is needed when the slave is actually shutting down. 

My proposal would be to remove this entirely- wdyt?